worker/controlleragentconfig/manifold.go

```diff
@@ -54,3 +60,12 @@ func Manifold(config ManifoldConfig) dependency.Manifold {
 func configOutput(in worker.Worker, out any) error {
 	return nil
 }
+
+// Notify sets up the signal handler for the worker.
+func Notify(ctx context.Context, ch chan os.Signal) {
+	if ctx.Err() != nil {
```

Comment:
this `ctx` is only needed to check if it's in error? shouldn't we check for `Done()` as well?

Reply:
`Err()` is a proxy for done in this case.

See the documentation:

```
	// If Done is not yet closed, Err returns nil.
	// If Done is closed, Err returns a non-nil error explaining why:
	// Canceled if the context was canceled
	// or DeadlineExceeded if the context's deadline passed.
	// After Err returns a non-nil error, successive calls to Err return the same error.
```

Reply:
oh right! thanks